File: dSoak-Common-Java/src/MessageTester/ThrowBalloonTester.java
package MessageTester;

import static org.junit.Assert.*;
import java.io.IOException;
import java.security.NoSuchAlgorithmException;
import org.junit.Test;
import Messages.Message;
import Messages.ThrowBalloon;
import SharedObject.Balloon;
import SharedObject.MessageNumber;

public class ThrowBalloonTester {

	@Test
	public void test_EveryThings() throws NoSuchAlgorithmException, IOException, ClassNotFoundException 
	{
		MessageNumber.LocalProcessId = 100;
		
		ThrowBalloon msg1 = new ThrowBalloon();
		assertNotNull(msg1.MessageNr);
		assertEquals(100, msg1.MessageNr.ProcessId);
		assertTrue(msg1.MessageNr.SeqNumber > 0);
		assertEquals(msg1.MessageNr, msg1.ConvId);
		
		Balloon b = new Balloon();
		b.Id = 245;
		ThrowBalloon msg2 = new ThrowBalloon();
		msg2.GameId  =123;
		msg2.Balloon = b;
		msg2.TargetPlayerId = 352;
		
		assertNotNull(msg2.MessageNr);
		assertNotNull(msg2.Balloon);
		assertEquals(100, msg2.MessageNr.ProcessId);
		assertTrue(msg2.MessageNr.SeqNumber > 0);
		assertEquals(msg1.MessageNr.SeqNumber + 1, msg2.MessageNr.SeqNumber);
		assertEquals(msg1.MessageNr.ProcessId, msg2.MessageNr.ProcessId);
		assertEquals(msg1.ConvId.ProcessId, msg2.ConvId.ProcessId);
		assertEquals(msg1.ConvId.SeqNumber + 1, msg2.ConvId.SeqNumber);
		assertEquals(123, msg2.GameId);
		assertEquals(352, msg2.TargetPlayerId);
		
		byte[] bytes = msg2.Encode();
		String str = new String(bytes);
		
		Message msg3 = Message.Decode(bytes);
		assertNotNull(msg3);
		assertEquals(msg2.MessageNr.ProcessId, msg3.MessageNr.ProcessId);
		assertEquals(msg2.MessageNr.SeqNumber, msg3.MessageNr.SeqNumber);
	
		
		ThrowBalloon msg4 = (ThrowBalloon) msg3;
		assertTrue(msg4 instanceof ThrowBalloon);
		assertEquals(msg3.MessageNr, msg4.MessageNr);
		assertEquals(msg3.ConvId, msg4.ConvId);
		
		assertEquals(msg2.MessageNr.ProcessId, msg4.MessageNr.ProcessId);
		assertEquals(msg2.MessageNr.SeqNumber, msg4.MessageNr.SeqNumber);
		assertEquals(msg2.ConvId.ProcessId, msg4.ConvId.ProcessId);
		assertEquals(msg2.ConvId.SeqNumber, msg4.ConvId.SeqNumber);
		assertEquals(msg2.TargetPlayerId, msg4.TargetPlayerId);
		assertEquals(msg2.Balloon.Id, msg4.Balloon.Id);
		assertNotNull(msg4.Balloon);
		
	}

}

File: dSoak-Common-CSharp/MessageTesting/ThrowBalloonTester.cs
using System;
using System.Collections.Generic;
using System.Text;
using Microsoft.VisualStudio.TestTools.UnitTesting;

using Messages;
using SharedObjects;

namespace MessageTesting
{
    [TestClass]
    public class ThrowBalloonTester
    {
        [TestInitialize]
        public void Setup()
        {
            MessageNumber.LocalProcessId = 100;
        }

        [TestMethod]
        public void ThrowBalloon_CheckEveything()
        {
            ThrowBalloon msg1 = new ThrowBalloon();
            Assert.IsNotNull(msg1.MessageNr);
            Assert.AreEqual(100, msg1.MessageNr.ProcessId);
            Assert.IsTrue(msg1.MessageNr.SeqNumber > 0);
            Assert.AreEqual(msg1.MessageNr, msg1.ConvId);

            Balloon b = new Balloon() { Id = 245 };
            ThrowBalloon msg2 = new ThrowBalloon() { GameId =123, Balloon = b, TargetPlayerId = 352 };
            Assert.IsNotNull(msg2.MessageNr);
            Assert.AreEqual(100, msg2.MessageNr.ProcessId);
            Assert.AreEqual(msg1.MessageNr.SeqNumber + 1, msg2.MessageNr.SeqNumber);
            Assert.AreEqual(msg2.MessageNr, msg2.ConvId);
            Assert.AreEqual(123, msg2.GameId);
            Assert.IsNotNull(msg2.Balloon);
            Assert.AreEqual(352, msg2.TargetPlayerId);

            byte[] bytes = msg2.Encode();
            string tmp = Encoding.ASCII.GetString(bytes);

            Message msg3 = Message.Decode(bytes);
            Assert.IsTrue(msg3 is ThrowBalloon);
            ThrowBalloon msg4 = msg3 as ThrowBalloon;
            Assert.AreEqual(msg2.MessageNr, msg4.MessageNr);
            Assert.AreEqual(msg2.ConvId, msg4.ConvId);
            Assert.AreEqual(msg2.GameId, msg4.GameId);
            Assert.IsNotNull(msg4.Balloon);
            Assert.AreEqual(msg2.Balloon.Id, msg4.Balloon.Id);
            Assert.AreEqual(msg2.TargetPlayerId, msg4.TargetPlayerId);
        }
    }
}
